packages/babel-plugin-transform-destructuring/src/util.ts

```diff
@@ -663,9 +659,32 @@ export function convertVariableDeclaration(
     if (!node.loc) {
       node.loc = nodeLoc;
     }
-    nodesOut.push(
-      inForInit && node.type === "ExpressionStatement" ? node.expression : node,
-    );
+    nodesOut.push(node);
+  }
+
+  if (
+    nodesOut.length === 2 &&
+    t.isVariableDeclaration(nodesOut[0]) &&
+    t.isExpressionStatement(nodesOut[1]) &&
+    t.isCallExpression(nodesOut[1].expression) &&
+    nodesOut[0].declarations.length === 1
+  ) {
```

Comment:
```suggestion
  if (
    nodesOut.length === 2 &&
    t.isVariableDeclaration(nodesOut[0]) &&
    t.isExpressionStatement(nodesOut[1]) &&
    t.isCallExpression(nodesOut[1].expression) &&
    nodesOut[0].declarations.length === 1
  ) {
    // This can only happen when we generate this code:
    //    var _ref = DESTRUCTURED_VALUE;
    //     babelHelpers.objectDestructuringEmpty(_ref);
    // Since pushing those two statements to the for loop .init will require an IIFE,
    // we can optimize them to
    //     babelHelpers.objectDestructuringEmpty(DESTRUCTURED_VALUE);
```